src/stac_asset/_functions.py

```diff
@@ -74,12 +74,19 @@ class Downloads:
     clients: Clients
     config: Config
     downloads: List[Download]
+    semaphore: asyncio.Semaphore
 
-    def __init__(self, config: Config, clients: Optional[List[Client]] = None) -> None:
+    def __init__(
+        self,
+        config: Config,
+        clients: Optional[List[Client]] = None,
+        max_concurrent_downloads: int = 1024,
```

Comment:
```suggestion
        max_concurrent_downloads: int = 500,
```

Per @jkeifer's suggestion.